src/main/java/org/myrobotlab/service/Gpt3.java

```diff
@@ -99,12 +109,28 @@ public Response getResponse(String text) {
         @SuppressWarnings({ "unchecked", "rawtypes" })
         Map<String, Object> payload = (Map) CodecUtils.fromJson(msg, LinkedHashMap.class);
         @SuppressWarnings({ "unchecked", "rawtypes" })
+        
+        Map<String,Object> errors = (Map)payload.get("error");
+        if (errors != null) {          
+          error((String)errors.get("message"));          
+        }        
+        @SuppressWarnings({ "unchecked", "rawtypes" })
         List<Object> choices = (List) payload.get("choices");
         if (choices != null && choices.size() > 0) {
           @SuppressWarnings({ "unchecked", "rawtypes" })
           Map<String, Object> textObject = (Map) choices.get(0);
           responseText = (String) textObject.get("text");
-          invoke("publishText", responseText);
+          if (responseText != null) {
+            // /completions
+            invoke("publishText", responseText);
+          } else {
+            // /chat/completions
+            @SuppressWarnings({ "unchecked", "rawtypes" })
+            Map<String, Object> content = (Map)textObject.get("message"); 

```

Comment:
You can cast to generic type, like so:
```java
content = (Map<String, Object>) textObject.get("message");
```

I still vote for a POJO based approach but this at least would get rid of using raw types

Reply:
still gives you a warning then for unchecked ... yeah POJOs a later pr